pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/text/LuceneTextIndexReader.java

```diff
@@ -70,6 +76,10 @@ public LuceneTextIndexReader(String column, File segmentIndexDir) {
       // Disable Lucene query result cache. While it helps a lot with performance for
       // repeated queries, on the downside it cause heap issues.
       _indexSearcher.setQueryCache(null);
+      // TODO: consider using a threshold of num docs per segment to decide between building
+      // mapping file upfront on segment load v/s on-the-fly during query processing
+      _docIdReaderWriter = new DocIdReaderWriter(segmentIndexDir, _column, numDocs);
+      _docIdReaderWriter.buildDocIdMapping(numDocs);
```

Comment:
We build the mapping every time we load the index? You saved the mapping into a file right?

Reply:
We check for existence and return.

Reply:
Also, I explored doing this in TextIndexHandler. But that is not good since it requires to open the lucene index, create searcher twice (both in handler and here anyway for query processing). I think it is better to avoid that and just open the lucene index reader and search just once per index.